File: scala/Fizzy.scala
#!/usr/bin/env scala

object Fizzy {
  def fizzy: Seq[String] = (1 to 100).par.map { n =>
    (n % 3, n % 5) match {
      case (0, 0) => "FizzBuzz"
      case (0, _) => "Fizz"
      case (_, 0) => "Buzz"
      case (_, _) => n.toString
    }
  }.seq

  def main(args: Array[String]): Unit = fizzy.foreach(println)
}

File: node/fizzy.js
#!/usr/bin/env node

function fizzBuzz(n) {
  var
  a = n % 3 == 0,
  b = n % 5 == 0;

  if (a && b) {
    return "FizzBuzz";
  }
  else if (a) {
    return "Fizz";
  }
  else if (b) {
    return "Buzz";
  }
  else {
    return n.toString();
  }
}

function main() {
  for (var i = 1; i < 101; i++) {
    console.log(fizzBuzz);
  }
}

if (!module.parent) { main(); }
